plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -1150,3 +1150,91 @@ FROM (
 	ORDER BY [record_id] DESC
 ) AS z
 `
+
+const sqlServerAvailabilityReplicaStates string = `
+IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) OR SERVERPROPERTY('IsHadrEnabled') = 0 BEGIN /*NOT IN Standard,Enterpris,Express*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:'+ @@ServerName + ',Database:' + DB_NAME() +' is not a SQL Server Standard,Enterprise or Express. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+
+SELECT
+	'sqlserver_hadr_avreplica_states' AS [measurement],
+	REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
+	convert(nvarchar(36), hars.replica_id) as replica_id,
+	ar.replica_server_name,
+	convert(nvarchar(36), hars.group_id) as group_id,
+	ag.name AS group_name,
+	ar.replica_metadata_id,
+	availability_mode,
+	availability_mode_desc,
+	failover_mode,
+	failover_mode_desc,
+	session_timeout,
+	primary_role_allow_connections,
+	primary_role_allow_connections_desc,
+	secondary_role_allow_connections,
+	secondary_role_allow_connections_desc,
+	seeding_mode,
+	seeding_mode_desc,
+	basic_features,
+	is_distributed,
+	is_local,
+	role,
+	role_desc,
+	operational_state,
+	operational_state_desc,
+	connected_state,
+	connected_state_desc,
+	recovery_health,
+	recovery_health_desc,
+	synchronization_health,
+	synchronization_health_desc,
+	last_connect_error_number,
+	last_connect_error_description,
+	last_connect_error_timestamp
+from sys.dm_hadr_availability_replica_states AS hars
+inner join sys.availability_replicas AS ar on hars.replica_id = ar.replica_id
+inner join sys.availability_groups AS ag on ar.group_id = ag.group_id
```

Comment:
Hi, very nice idea :) May I suggest to add info about AG state in this query (or maybe in a different query to keep consistent) ? :

```
SELECT
	'sqlserver_hadr_avreplica_states' AS [measurement],
	REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
	convert(nvarchar(36), hars.replica_id) as replica_id,
	ar.replica_server_name,
	convert(nvarchar(36), hars.group_id) as group_id,
	ag.name AS group_name,
	ag.basic_features,
	ag.is_distributed,
	hags.synchronization_health_desc AS ag_synchronization_health_desc,
	ar.replica_metadata_id,
	ar.availability_mode ,
	ar.availability_mode_desc,
	ar.failover_mode,
	ar.failover_mode_desc,
	ar.session_timeout,
	ar.primary_role_allow_connections,
	ar.primary_role_allow_connections_desc,
	ar.secondary_role_allow_connections,
	ar.secondary_role_allow_connections_desc,
	ar.seeding_mode,
	ar.seeding_mode_desc,
	hars.is_local,
	hars.role,
	hars.role_desc,
	hars.operational_state,
	hars.operational_state_desc,
	hars.connected_state,
	hars.connected_state_desc,
	hars.recovery_health,
	hars.recovery_health_desc,
	hars.synchronization_health AS replica_synchronization_health,
	hars.synchronization_health_desc AS replica_synchronization_health_desc,
	hars.last_connect_error_number,
	hars.last_connect_error_description,
	hars.last_connect_error_timestamp
from sys.dm_hadr_availability_replica_states AS hars
inner join sys.availability_replicas AS ar on hars.replica_id = ar.replica_id
inner join sys.availability_groups AS ag on ar.group_id = ag.group_id
INNER JOIN sys.dm_hadr_availability_group_states AS hags ON hags.group_id = ag.group_id
```

Reply:
I am not for having a separate query, which means yet another one to run in an iteration. @avinash-nigam  if adding group state you may have to name the columns appropriately for state :)

One more thing @avinash-nigam  - can we rename the measurement to sqlserver_hadr_replica_states ( no need for the av part?)

Thanks much!

Reply:
@mikedavem included AG state as per your and Denzil's suggestions
@denzilribeiro removed the "av" part from the measurement